src/main/java/TaskListener.java

```diff
@@ -0,0 +1,241 @@
+        this.tasksList = tasksList;
+        this.tasksLoader = tasksLoader;
+    }
+
+    public static void greet() {
+        String logo =
+                        "   _____ __  ____________  __    ____  ________ __\n" +
+                        "  / ___// / / / ____/ __ \\/ /   / __ \\/ ____/ //_/\n" +
+                        "  \\__ \\/ /_/ / __/ / /_/ / /   / / / / /   / ,<   \n" +
+                        " ___/ / __  / /___/ _, _/ /___/ /_/ / /___/ /| |  \n" +
+                        "/____/_/ /_/_____/_/ |_/_____/\\____/\\____/_/ |_|  ";
+
+
+        printLines("Hello from", logo);
+
+        printLines("Hello! I'm SHERLOCK", "What can I do for you?");
+    }
+
+    public static void printLines(String... lines) {
+        System.out.println(DIVIDER);
+        for (String line : lines) {
+            System.out.println(line);
+        }
+        System.out.println(DIVIDER + System.lineSeparator());
+    }
+    private void printAddedTask(Task task) {
+        String tasksWord = tasksList.getTasksCount() == 1 ? " task" : " tasks";
+        final String FIRST_SENTENCE = "Got it. I've added this task:";
+        final String SECOND_SENTENCE = "Now you have " + tasksList.getTasksCount() + tasksWord + " in the list.";
+        printLines(FIRST_SENTENCE + System.lineSeparator() + task + System.lineSeparator() + SECOND_SENTENCE);
+    }
+
+    private void addGenericTask(String name) {
+        if (name.isEmpty()) {
+            printLines("Please provide task description");
+            return;
+        }
+        this.tasksList.addTask(new Task(name, false));
+
+        printLines("added: " + name);
+
+        tasksLoader.writeToFile(tasksList);
+    }
+
+    private void createTodo(String arguments) {
+        String name = arguments;
+        if (name.isEmpty()) {
+            printLines("Please provide todo description");
+            return;
+        }
+        Todo todo = new Todo(name, false);
+        this.tasksList.addTask(todo);
+        printAddedTask(todo);
+
+        tasksLoader.writeToFile(tasksList);
+    }
+
+    private void createDeadline(String arguments) {
+        int byCommandIndex = arguments.indexOf("/by");
+        if (byCommandIndex < 0) {
+            printLines("Please specify the /by command");
+            return;
+        }
+        String name = arguments.substring(0, byCommandIndex).trim();
+        String byArgumentValue = arguments.substring(byCommandIndex).substring("/by".length()).trim();
+
+        if (name.isEmpty()) {
+            printLines("Please provide name argument");
+            return;
+        }
+
+        if (byArgumentValue.isEmpty()) {
+            printLines("Please provide /by argument");
+            return;
+        }
+        Deadline deadline = new Deadline(name, false, byArgumentValue);
+        this.tasksList.addTask(deadline);
+
+        printAddedTask(deadline);
+
+        tasksLoader.writeToFile(tasksList);
+    }
```

Comment:
Maybe you should divide into smaller methods since this methods seems to be abit long same with createEvent method. And to handle exception, you might consider using Exception instead of print out the error. It would beneficial for your tP when you have to handle a lot of errors